test/server/admin/admin_test.cc

```diff
@@ -217,6 +222,242 @@ TEST_P(AdminInstanceTest, ConfigDumpMaintainsOrder) {
+  ON_CALL(server_.cluster_manager_, clusters()).WillByDefault(ReturnPointee(&cluster_map));
+
+  NiceMock<Upstream::MockClusterMockPrioritySet> cluster;
+  cluster_map.emplace(cluster.info_->name_, cluster);
+
+  ON_CALL(*cluster.info_, addedViaApi()).WillByDefault(Return(false));
+
+  Upstream::MockHostSet* host_set = cluster.priority_set_.getMockHostSet(0);
+  auto host = std::make_shared<NiceMock<Upstream::MockHost>>();
+
+  envoy::config::core::v3::Locality locality;
+  ON_CALL(*host, locality()).WillByDefault(ReturnRef(locality));
+
+  host_set->hosts_.emplace_back(host);
+  Network::Address::InstanceConstSharedPtr address =
+      Network::Utility::resolveUrl("tcp://1.2.3.4:80");
+  ON_CALL(*host, address()).WillByDefault(Return(address));
+  const std::string hostname = "foo.com";
+  ON_CALL(*host, hostname()).WillByDefault(ReturnRef(hostname));
+
+  auto metadata = std::make_shared<envoy::config::core::v3::Metadata>();
+  ON_CALL(*host, metadata()).WillByDefault(Return(metadata));
+
+  const std::string hostname_for_health_checks = "test_hostname_healthcheck";
+  ON_CALL(*host, hostnameForHealthChecks()).WillByDefault(ReturnRef(hostname_for_health_checks));
+  Network::Address::InstanceConstSharedPtr healthcheck_address =
+      Network::Utility::resolveUrl("tcp://1.2.3.5:90");
+  ON_CALL(*host, healthCheckAddress()).WillByDefault(Return(healthcheck_address));
+
+  ON_CALL(*host, health()).WillByDefault(Return(Upstream::Host::Health::Healthy));
+
+  ON_CALL(*host, weight()).WillByDefault(Return(5));
+  ON_CALL(*host, priority()).WillByDefault(Return(6));
+
+  Buffer::OwnedImpl response;
+  Http::TestResponseHeaderMapImpl header_map;
+  EXPECT_EQ(Http::Code::OK, getCallback("/config_dump?includeEds", header_map, response));
+  std::string output = response.toString();
+  const std::string expected_json = R"EOF({
+ "configs": [
+  {
+   "@type": "type.googleapis.com/envoy.admin.v3.EndpointsConfigDump",
+   "static_endpoint_configs": [
+    {
+     "endpoint_config": {
+      "@type": "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment",
+      "cluster_name": "fake_cluster",
+      "endpoints": [
+       {
+        "locality": {},
+        "lb_endpoints": [
+         {
+          "endpoint": {
+           "address": {
+            "socket_address": {
+             "address": "1.2.3.4",
+             "port_value": 80
+            }
+           },
+           "health_check_config": {
+            "port_value": 90,
+            "hostname": "test_hostname_healthcheck"
+           },
+           "hostname": "foo.com"
+          },
+          "health_status": "HEALTHY",
+          "metadata": {},
+          "load_balancing_weight": 5
+         }
+        ],
+        "priority": 6
+       }
+      ],
+      "policy": {
+       "overprovisioning_factor": 140
+      }
+     }
+    }
+   ]
+  }
+ ]
+}
+)EOF";
+  EXPECT_EQ(expected_json, output);
+}
+
+// Test EDS config dump while locality and locality weight exists
```

Comment:
why do you need separate tests for locality and locality weight?

Reply:
Endpoints which belong to the same locality need to be generated in the same `locality_lb_endpoint`. The process is different from the situation that no endpoint belongs to one locality and is implemented seprately.
This is implemented by `if (!host_set->hostsPerLocality().get().empty())` in [admin.cc](https://github.com/envoyproxy/envoy/pull/11577/files/06ca89100033acdb3364509765b824dbff99efef#diff-d48056c908b9f9aa84de73a7d180bd13)
Adding this test just makes sure the coverage of unit test.